### DISCUSSION TITLE
Upgraded to EdgeTX 2.8.0, radio freezes on 2nd boot

### DISCUSSION THREAD

**Discussion by Alan-70** [EdgeTX/edgetx#3095] [Q&A]
I’m an ancient noob to modern radio control.  I’ve installed EdgeTx 2.8.0 to my RadioMaster TX16SMk2 Max, having bought it with 2.7.1 installed. I think EdgeTx is fantastic and I’m just beginning to understand its potential. The Developers are doing an amazing job and I’m quite sure my problem is my fault!

I’ve successfully flashed 2.8.0 along with v1.3.3.20.bin of the 4-in-1 MPM.

I’ve done a full format of the 32Gb SD card, then copied in the contents of the new SD card C480x272, and the latest SOUNDS folder.

I’ve then copied in the contents of my 2.7.1 MODELS, RADIO &  IMAGES folders.

I put the SD card back in the radio, switched on, got ‘Welcome to EdgeTx’ message. Everything worked fine, firmware version correct, my receiver still bound etc.

PROBLEM: Then I switched off and when I switched on again it hung on the splash screen, didn’t respond to any key presses, and I had to unplug the battery to get rid of the splash screen (it then went into Emergency Mode on the next boot). The screen freeze/battery disconnect repeats.

If I put in the 2.7.1 SD card (but with the TX still flashed to 2.8.0 of course) it runs perfectly well.

Maybe I should let it lie, but there are of course some differences between the SD contents for 2.80 and 2.7.1 and anyway it’s driving me mad. I don't want to get out of sync between Edge Tx version and the SD card. What can I be doing wrong?

Many thanks if you can help.

Alan

**Comment by pfeerick** [comment]
Whether you are using 2.7.1 or 2.8.0 SD card pack is of little importance - EdgeTX is not as dependent on the SD card "pack" version as OpenTX was, as all the core resources (graphics, icons, splash screen, etc) are built into the firmware. The fact that it is reproducible by simply swapping the card suggests there is some incompatibility with the SD card itself. Double check the 32GB card is formatted as FAT/FAT32 (not exFAT!!) and then try putting the 2.7.1 SD card content on as a comparision - I'll bet it still plays up.

**Reply by Alan-70** [reply]
Hi Peter, thanks very much for taking the time. that's very helpful.

So I could just copy in to my (running OK)  2.7.1 card the missing 2.8.0 SD card folders (eg SoarETX which I'm really interested in)?

And your suspicion then is that there might be something dodgy with my (brand new 32Gb) SD card? 

I'll do as you suggest (put the 2.7.1 SD card contents on) and see what happens.

Alan

**Reply by Alan-70** [reply]
Well, this gets weirder!

So it's running fine with the 2.7.1 SD card and firmware 2.8.0 on the 32 Gb card.

Using the PC with the SD card in teh card reader slot, I deleted the 2.7.1 'WIDGETS' folder and replaced it with the 2.8.0 SD Card WIDGETS.

The radio started its 'first time boot OK, 2nd time stuck' routine.

This before I've actually tried to run any of the new widgets (in 2.8.0 4 of the 2.71. widgets are gone, and there are 3 new ones). 

I am though running in 2.7.1 widgets to show the model name and image, outputs, and timer 1.

I can't see how one of the new widgets can be mucking things up, since I haven't called any of them.

Suggesting that on the 2.8.0 SD card something has changed in the widgets which run model/outputs/timer?

I'm baffled.

Alan

**Reply by pfeerick** [reply]
Ok, yes, that is where you could come unstuck... newer versions of widgets could be relying on Lua API calls that don't exist in an older version of the firmware... but generally, that should simply make them crash internally and get killed off, not lock up the radio in any sense. Will have to look into that - won't be for a day or two though.

**Reply by Alan-70** [reply]
Peter, many thanks indeed. In case its of any help, this morning I did the following: in bootloader mode using Companion wrote models and settings to the PC (one model file corrupt so I deleted it) and saved a copy of IMAGES;  full format on the SD card (FAT 32); extracted 2.8.0 SD card contents to the blank SD card, on the PC; copied images back; in bootloader mode used Companion to write models and settings back to the radio. Exited bootloader, switched on, radio stuck as before. Replaced with the other SD card holding 2.7.0 SD card data, all OK. Incidentally ithe radio shows 'bootloader 2.7.1' with 'edgetx 2.8.0' at bottom of screen.
I'm just going to nip into town and buy another SD card and do it all over again, just in case it is a dodgy card.
Thanks again for your help, much appreciated. 
Alan

**Reply by Alan-70** [reply]
Peter, just to confirm that I've now acquired a new 16Gb SD card, religiously followed 'RC Video Reviews - How to Upgrade to a new SD card' and same result - first time radio is switched on all is well, second time it freezes at the splash screen. So its not an SD card failure.
Alan

**Reply by Alan-70** [reply]
Peter, I can provide more diagnostic info. Please say if you want me to raise this as an 'issue':
1. Using full reformatted card, extracted all of C480x272.zip to it
2. Added no model or radio files (or the latest sound files)
3. Inserted in Tx16s Mk2, power on, 'bad radio', press 'enter', set up radio afresh
4. Calibrated sticks etc. ok
5. Repeated calibration and unable to exit calibration routine ('enter' key unresponsive)
6. set up  simple model, ok, bind to rx, ok
7. power off
8. power on, 'Throttle warning ....press any key', no response from any key
9. battery out, restart with throttle at -100, still got throttle warning, but 'any key' cleared it
10. mucked about with mixes etc on the one model
11. power off
12. power on - stuck on splash screen

When I swapped the SD card for the one I've had since 2.7.1 everything works fine.

Hope this is helpful.

regards
Alan

**Reply by Alan-70** [reply]
Hi Peter, I just wanted to let you know that I think this issue is solved, following some extensive coaching from Risto over on RCGroups. Eventually I reflashed the radio using STM32CubeProgrammer and a fresh copy of the 2.8.0 firmware binary, and now (touch wood) all seems well. I also used EdgeTx buddy to make a new SD card - very impressively easy software. I don't think we'll ever know how I got in the mess in the first place! I'm very grateful for your help.
regards
Alan
